numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2092,83 +2091,101 @@ get_single_op_view(PyArrayObject *op, int  iop, char *labels,
     return 0;
 }
 
+static int
+_no_labels_are_negative(signed char *labels, int ndim)
```

Comment:
I'd maybe prefer `_any_labels_are_negative`, and then use it with a preceding `!`. Functions with inverted names are a little confusing - also `any` is common python terminology